### DISCUSSION TITLE
Configure workspaces on launch

### DISCUSSION THREAD

**Discussion by jamesredwards** [wezterm/wezterm#6238] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Sway

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I'm trying to set my config to open up some default workspaces on launch using the below code. If I use "gui-startup" rather than "mux-startup" each workspace opens in a separate instance of wezterm which isn't what i'm after. The below doesn't actually work at all so I'm not sure what I'm doing wrong. Clearly have not understood something in the docs!

```lua
wezterm.on("mux-startup", function(cmd)
	local mux = wezterm.mux
	local rust_dir = "~/programming/rust/"
	local c_dir = "~/programming/c/"
	local py_dir = "~/programming/python/"
	local dot_dir = "~/programming/dotfiles/"

	local args = {}
	if cmd then
		args = cmd.args
	end

	local rust_workspace = mux.spawn_window({
		workspace = "rust",
		cwd = rust_dir,
	})
	local c_workspace = mux.spawn_window({
		workspace = "c",
		cwd = c_dir,
	})
	local dots_workspace = mux.spawn_window({
		workspace = "dots",
		cwd = dot_dir,
	})
	local py_workspace = mux.spawn_window({
		workspace = "python",
		cwd = py_dir,
	})
	mux.set_active_workspace("rust")
end)
```

**Comment by feoh** [comment]
I bet you want something like what's detailed [here](https://github.com/wez/wezterm/issues/3646#issuecomment-1529022385).

```lua
local wezterm = require "wezterm"
local mux = wezterm.mux

wezterm.on('gui-startup', function()
  local _, first_pane, window = mux.spawn_window {}
  local _, second_pane, _ = window:spawn_tab {}
  local _, third_pane, _ = window:spawn_tab {}

  second_pane:send_text "<insert busybox command>\n"
  third_pane:send_text "top\n"
  -- '\n' this will execute you shell command
end)
```

**Comment by estin** [comment]
May be https://github.com/wez/wezterm/discussions/6477 to configure workspaces and select them
